### PR TITLE
data model for podcasts

### PR DESCRIPTION
**Pull request by abeglova**
#### Pre-Flight checklist

- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Changes have been manually tested

#### What are the relevant tickets?
https://github.com/mitodl/open-discussions/issues/2735

#### What's this PR do?
Adds `Podcast` and `PodcastEpisode` for podcast ingestion

#### How should this be manually tested?
```python
from course_catalog.factories import *
p = PodcastFactory.create()
pe = PodcastEpisodeFactory.create()
```

### CODE REVIEW

**Comment by rhysyngsun** [comment]
also has a conflict that need to be resolved

File: course_catalog/models.py [hunk 1]
```diff
@@ -472,3 +472,40 @@ class PlaylistVideo(models.Model):
...
+
+    full_description = models.TextField(null=True, blank=True)
+    image_src = models.URLField(max_length=400, null=True, blank=True)
+    last_modified = models.DateTimeField(null=True, blank=True)
+    platform = models.CharField(max_length=128)
+    podcast = models.ForeignKey(
+        Podcast, related_name="episodes", on_delete=models.CASCADE
+    )
+    published = models.BooleanField(default=True)
+    runs = GenericRelation(LearningResourceRun)
```
[the remark this answers is not in the document]
Comment (abeglova) [reply course_catalog/models.py:506]: Why did you end up adding runs to videos? I think it was something along the lines of runs being required for the "free" filter to work properly

**Comment by abeglova** [comment]
Thanks @rhysyngsun back to you

File: course_catalog/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rhysyngsun) [reply course_catalog/models.py:527]: yeah probably, :-\ i guess leave it then

File: course_catalog/models.py [hunk 2]
```diff
@@ -472,3 +489,51 @@ class PlaylistVideo(models.Model):
...
+    full_description = models.TextField(null=True, blank=True)
+    image_src = models.URLField(max_length=400, null=True, blank=True)
+    published = models.BooleanField(default=True)
+    url = models.URLField(null=True, max_length=2048)
+    searchable = models.BooleanField(default=True)
+
+    @property
+    def platform(self):
+        """Platform for podcasts"""
+        return PlatformType.podcast
```
[the remark this answers is not in the document]
Comment (abeglova) [reply course_catalog/models.py]: Good catch. Thanks!

**Pull request merged**